inst/components/sidebar.scss

```
@@ -316,21 +316,31 @@
 }
 
 
-// In the case of page_navbar(nav_panel("Foo", layout_sidebar())), we want the
-// sidebar to be flush with the viewport (just like it will be in the
-// case of page_navbar(sidebar = sidebar()))
 .navbar + .container-fluid {
-  &:has(>.tab-content>.tab-pane.active>.bslib-sidebar-layout) {
+  // When the tab/page is fillable, undo the padding on the container-fluid
+  // (in this case, the user has control over padding/gap)
+  &:has(> .tab-content > .tab-pane.active.html-fill-container) {
     padding-left: 0;
     padding-right: 0;
   }
-  > .tab-content > .tab-pane.active > .bslib-sidebar-layout {
-    &:not([data-bslib-sidebar-border="true"]) {
-      border: none;
-    }
-    &:not([data-bslib-sidebar-border-radius="true"]) {
-      border-radius: 0;
+  > .tab-content > .tab-pane.active {
+
+    // When the tab/page is fillable, add sensible default padding
+    &.html-fill-container {
+      padding: $spacer;
+      &:has(> .bslib-sidebar-layout) {
+        padding: 0;
+      }
+      // And smart border defaults for nav_panel("Foo", layout_sidebar())
+      > .bslib-sidebar-layout.html-fill-container {
+        &:not([data-bslib-sidebar-border="true"]) {
+          border: none;
```

```suggestion
          border-left: none;
          border-right: none;
          border-bottom: none;
```

We should do something like this to make this situation better?

```r
page_navbar(inverse = FALSE, nav_panel("Page 1", layout_sidebar()))
```

---

If `inverse = FALSE` here is about the navbar style, I'd personally vote for not shipping that yet and in a (near) future release borrowing the ideas from quarto to make navbar style easier to customize. (In other words I think we'll outgrow a binary option.)

---

`inverse` was shipped along time ago...

---

Ohhh 🙊 please ignore me then 🙂